lib/tasks/sitemap.rake

```diff
@@ -0,0 +1,13 @@
+
+namespace :sitemap do
+  desc "Generate sitemap.xml and sitemap.xml.gz files"
+  task :generate, [ :base_url ] => :environment do |task, args|
+    sitemap = Sitemap.new(base_url: args[:base_url], update_robots: true)
```

Comment:
Consider adding error handling for cases where the base URL is malformed beyond what the validation catches, or when the rake task fails.